### PR TITLE
[windows][depends] Update dependencies for all windows platforms

### PR DESCRIPTION
**Pull request by Paxxi**
Updates most dependencies to latest version
Fixes the store build by fixing python
Enables pillow and pycryptodome again on windows
Replaces easyhook with detours

This also adds curl support for brotli and http2. detours is tested on win64 that it correctly hooks the methods.

Everything is built from https://github.com/Paxxi/kodi-deps/tree/work . To replicate a build install the required tools described in the readme and then run `DoRelease.ps1 -Desktop -App -VsVersion 15` to build all platforms and create zip files.

There's also a full set of dependencies for arm64 and arm64 store on the ftp if anyone want to get that working without having to build everything themselves 😄 

## Motivation and Context

unblock v19 release

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: cmake/scripts/windows/ArchSetup.cmake [hunk 1]
```diff
@@ -45,12 +45,9 @@ set(SYSTEM_DEFINES -DWIN32_LEAN_AND_MEAN -DNOMINMAX -DHAS_DX -D__STDC_CONSTANT_M
                    -DPLT_HTTP_DEFAULT_USER_AGENT="UPnP/1.0 DLNADOC/1.50 Kodi"
                    -DPLT_HTTP_DEFAULT_SERVER="UPnP/1.0 DLNADOC/1.50 Kodi"
                    -DUNICODE -D_UNICODE
+                   -DFRIBIDI_STATIC
```
Comment (Rechi) [inline cmake/scripts/windows/ArchSetup.cmake]: this is the wrong place to add a define for a library

File: cmake/scripts/windowsstore/ArchSetup.cmake [hunk 2]
```diff
@@ -71,6 +71,7 @@ set(SYSTEM_DEFINES -DWIN32_LEAN_AND_MEAN -DNOMINMAX -DHAS_DX -D__STDC_CONSTANT_M
                    -DPLT_HTTP_DEFAULT_USER_AGENT="UPnP/1.0 DLNADOC/1.50 Kodi"
                    -DPLT_HTTP_DEFAULT_SERVER="UPnP/1.0 DLNADOC/1.50 Kodi"
                    -DUNICODE -D_UNICODE
+                   -DFRIBIDI_STATIC
```
Comment (Rechi) [inline cmake/scripts/windowsstore/ArchSetup.cmake]: this is the wrong place to add a define for a library

File: xbmc/guilib/TextureBundleXBT.cpp [hunk 3]
```diff
@@ -26,10 +26,8 @@
 #ifdef TARGET_WINDOWS_DESKTOP
 #ifdef NDEBUG
 #pragma comment(lib,"lzo2.lib")
-#elif defined _WIN64
-#pragma comment(lib, "lzo2d.lib")
 #else
-#pragma comment(lib, "lzo2-no_idb.lib")
+#pragma comment(lib, "lzo2d.lib")
```
Comment (Rechi) [inline xbmc/guilib/TextureBundleXBT.cpp]: This has to be handled by the build system.

File: xbmc/windowing/windows/WinSystemWin32DX.cpp [hunk 4]
```diff
@@ -20,15 +20,21 @@
 #include "system.h"
 
 #ifndef _M_X64
-#pragma comment(lib, "EasyHook32.lib")
 #include "utils/SystemInfo.h"
+#endif
+#if _DEBUG
+#pragma comment(lib, "detoursd.lib")
```
Comment (Rechi) [inline xbmc/windowing/windows/WinSystemWin32DX.cpp]: This has to be handled by the build system.

File: xbmc/windowing/windows/WinSystemWin32DX.cpp [hunk 5]
```diff
@@ -20,15 +20,21 @@
...
 #else
-#pragma comment(lib, "EasyHook64.lib")
+#pragma comment(lib, "detours.lib")
```
Comment (Rechi) [inline xbmc/windowing/windows/WinSystemWin32DX.cpp]: This has to be handled by the build system.

**Review by Rechi — COMMENTED** [review]
There are a few thing which aren't acceptable and I already told you in the previous PR, library define in ArchSetup.cmake  and `#pragma comment(lib, ...)`. Also the detours find module is missing Those have to be fixed.

Also why have you merged this PR (after only 17 hours) without waiting for a review?

**Comment by Paxxi** [comment]
It's needed to unblock v19 and move to alpha. I fixed the relevant comments such as removing the platform ifdef for crossguid. The remaining comments isn't relevant to this PR and can be handled in separate PRs to improve the build system.

**Comment by Rechi** [comment]
You added a new dependency detours and using the build system for linking is relevant for it.
Also it seems like didn't even try to remove the lzo2 pragma linking.

Fixing the build is no excuse to add hacks, so please fix it now.

**Comment by Paxxi** [comment]
I didn't add any hacks, I only used what was already there.
There's more important issues to fix that actually affect users such as python crashes that I'll spend my time on. Once that's sorted I might revisit this and clean up the build system but nobody has done it so far so it's not like there's any rush to get it done.

**Comment by Rechi** [comment]
`FRIBIDI_STATIC` and `#pragma comment(lib, "detours.lib")` define didn't exist before. Also you added a dependency without adding a find module, this is a hack too.

Before you start working on anything else fix this PR.

**Comment by Rechi** [comment]
@Paxxi what's the estimated time to expected the follow up for this PR?

**Comment by Paxxi** [comment]
There is no ETA, spent the time yesterday to look at python crash https://github.com/xbmc/xbmc/pull/17151 and will spend the coming days to look at the HDR PR as agreed upon during the team meating.

Once that's sorted I'll try and spend some time on cleaning things up.

**Comment by Essam311** [comment]
@Paxxi Could you fix it so the Netflix plugin be able to run without error, you do not need to have a Netflix account to see the error, just install the add-on. The error is pycryptodome related, I tested this nightly.

https://mirrors.kodi.tv/nightlies/windows/win64/master/KodiSetup-20200107-13b742ee-master-x64.exe

https://github.com/asciidisco/plugin.video.netflix/releases

**Comment by CastagnaIT** [comment]
log here: https://paste.kodi.tv/fuwovodizu.kodi

seem that some pyd files needed are missing in:
system\Python\Lib\site-packages\Cryptodome\Hash\

in this case _MD5, but compared to my local windows py installation, there are also others pyd missing

**Comment by Paxxi** [comment]
Thank you! Working on a fix

**Comment by Paxxi** [comment]
@CastagnaIT and @Essam311 can you test this build and see if it solves the issue for you? http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20200108-ebc4baf9-pycryptodome-x64.exe

**Comment by CastagnaIT** [comment]
much better now but missing another one pyd: _cpuid_c
Log: https://paste.kodi.tv/azitogejeq.kodi
(in the log do not take into account of Crypto module error, so ignore it)

**Comment by Paxxi** [comment]
Thanks again. _cpuid_c should be there as _cpuid Will fix that and get it merged

**Comment by CastagnaIT** [comment]
thanks to you

**Pull request merged**